network/p2p/libp2pNode.go

```diff
@@ -58,19 +60,27 @@ func DefaultLibP2PNodeFactory(log zerolog.Logger, me flow.Identifier, address st
 		pubsub.WithStrictSignatureVerification(false),
 		// set max message size limit for 1-k PubSub messaging
 		pubsub.WithMaxMessageSize(maxPubSubMsgSize),
+		// no discovery
+	}
+
+	options := []NodeOption{
+		WithDefaultLibP2PHost(address, connManager, flowKey, true),
+		WithDefaultPubSub(psOptions...),
+		WithDefaultPingService(rootBlockID, pingInfoProvider),
 	}
 
 	return func() (*Node, error) {
-		return NewLibP2PNode(log, me, address, NewConnManager(log, metrics), flowKey, true, rootBlockID, pingInfoProvider, psOptions...)
```

Comment:
The functional options pattern should be used to specify configure those options for which there are sensible defaults.

In this case, I don't think `address` or `rootBlockID` for example have any sensible default values, so they should be passed in explicitly.

Reply:
If we take libp2p's own library as an example, we can see that all the pre-defined [functional options](https://github.com/libp2p/go-libp2p-pubsub/blob/37d36d77b1953fedea26cb805ca2bd993b4b90b1/pubsub.go#L321-L457) only configure fields for which there are [default values](https://github.com/libp2p/go-libp2p-pubsub/blob/37d36d77b1953fedea26cb805ca2bd993b4b90b1/pubsub.go#L233-L274) (e.g `msgID`, `signPolicy`, etc)

Reply:
ahh...that makes sense! will fix.